applicationinsights-react-native/src/ReactNativePlugin.ts

```diff
@@ -48,34 +53,35 @@ export class ReactNativePlugin extends BaseTelemetryPlugin {
                 core?: IAppInsightsCore,
                 extensions?: IPlugin[]
             ) => {
+                let identifier = this.identifier;
                 if (!_self.isInitialized()) {
                     _base.initialize(config, core, extensions);
 
-                    const inConfig = config || {};
-                    const defaultConfig = _getDefaultConfig();
-                    objForEachKey(defaultConfig, (option, value) => {
-                        _config[option] = ConfigurationManager.getConfig(
-                            inConfig as any,
-                            option,
-                            _self.identifier,
-                            !isUndefined(_config[option]) ? _config[option] : value
-                        );
-                    });
-        
-                    if (!_config.disableDeviceCollection) {
-                        _self._collectDeviceInfo();
-                    }
-        
-                    if (core && core.getPlugin) {
-                        _analyticsPlugin = core.getPlugin<any>(AnalyticsPluginIdentifier)?.plugin as IAppInsights;
-                    }
-        
-                    if (!_config.disableExceptionCollection) {
-                        _self._setExceptionHandler();
-                    }
+                    _self._addHook(onConfigChange(config, (details) => {
+                        let ctx = _self._getTelCtx();
+                        _config = ctx.getExtCfg<IReactNativePluginConfig>(identifier, defaultReactNativePluginConfig);
+            
+                        if (!_config.disableDeviceCollection) {
+                            _self._collectDeviceInfo();
+                        }
+            
+                        if (core && core.getPlugin) {
+                            _analyticsPlugin = core.getPlugin<any>(AnalyticsPluginIdentifier)?.plugin as IAppInsights;
+                        } else {
+                            _analyticsPlugin = null;
+                        }
+            
+                        if (!_config.disableExceptionCollection) {
+                            _self._setExceptionHandler();
```

Comment:
We probably should have a "removeExceptionHandler" (if we previously added one) or we use the trick of returning an object from the _set function that has the `rm` function.